aggregator/src/datastore.rs

```diff
@@ -3375,7 +3332,7 @@ ORDER BY id DESC
                         FROM aggregate_share_jobs
                         JOIN aggregation_jobs
                             ON aggregation_jobs.task_id = aggregate_share_jobs.task_id
-                            AND (aggregation_jobs.batch_identifier = aggregate_share_jobs.batch_identifier
+                            AND (aggregation_jobs.batch_id = aggregate_share_jobs.batch_identifier
                               OR aggregation_jobs.client_timestamp_interval <@ aggregate_share_jobs.batch_interval)
```

Comment:
I think this may have to change to && as well. If all overlapping aggregation jobs end up with wider intervals than the interval of an aggregate share job, then that aggregate share job would always get filtered out.